src/tardev-snapshotter/docs/ROADMAP.md

```diff
@@ -0,0 +1,144 @@
+     IMAGE-LAYER-SIGNATURE-MANIFEST ||--|| IMAGE-LAYER-SIGNATURE: layers
+   ```
+
+### Milestone 2: Pre-Population of EROFS Tar Index and dm-verity Merkle Tree
+
+This milestone focuses on generating the EROFS tar index and dm-verity Merkle
+tree in advance, eliminating the need for on-the-fly generation on the container
+host. This approach accelerates cold container startup and reduces overall
+energy consumption by performing these operations once, rather than on every
+host.
+
+The erofs-snapshotter will be extended to fetch the tar index and dm-verity
+Merkle tree from the registry. If unavailable, the snapshotter will fall back to
+generating these artifacts locally, as described in Milestone 1.
+
+Implementation will leverage OCI referrers to loosely associate the tar index
+and dm-verity Merkle tree with the original image layers, similar to the
+signature mechanism established previously. This decouples artifact generation
+from image building, allowing post-publish addition of these artifacts to help
+with onboarding. It also ensures compatibility with other containerd
+snapshotters that do not support EROFS or dm-verity.
+
+Using OCI referrers, rather than modifying published registry artifacts, is a
+cleaner approach that avoids manifest regeneration.
+
+```mermaid
+erDiagram
+    direction LR
+    IMAGE-MANIFEST }|--|{ IMAGE-LAYER-TAR: layers
+    IMAGE-LAYER-TAR ||--|| IMAGE-LAYER-MANIFEST: layers
+    IMAGE-LAYER-MANIFEST ||..|{ IMAGE-LAYER-SIGNATURE-MANIFEST: subject
+    IMAGE-LAYER-SIGNATURE-MANIFEST ||--|| IMAGE-LAYER-SIGNATURE: layers
+    IMAGE-LAYER-MANIFEST ||..|| IMAGE-LAYER-CI-MANIFEST: subject
+    IMAGE-LAYER-CI-MANIFEST ||--|| IMAGE-LAYER-CI: layers
+    IMAGE-LAYER-MANIFEST ||..|| IMAGE-LAYER-EROFS-COMPATIBILITY-MANIFEST: subject
+    IMAGE-LAYER-EROFS-COMPATIBILITY-MANIFEST ||--|| IMAGE-LAYER-EROFS-COMPATIBILITY: layers
+```
+
+- `IMAGE-LAYER-CI`: Contains the dm-verity Merkle tree.
+- `IMAGE-LAYER-EROFS-COMPATIBILITY`: Contains the tar index.
+
+A potential drawback is the need for the snapshotter to fetch additional
+artifacts (tar index and Merkle tree) from the registry during initial layer
+preparation. To mitigate latency, collaboration is encouraged on [native
+referrer support in
+containerd](https://github.com/containerd/containerd/issues/7654), enabling
+pre-fetching of such artifacts during image pull operations. This would also
+benefit signature artifact retrieval.
+
+### Milestone 3: Native Support for EROFS + dm-verity Layers
+
+This milestone aims to deliver native support for EROFS and dm-verity layers in
+containerd/erofs-snapshotter, reducing performance overhead for teams capable of
+publishing and consuming native EROFS+dm-verity layers.
+
+Currently, containerd selects the appropriate image manifest based on OS and
+architecture via the image index. The plan is to extend this mechanism to
+consider OS features, allowing the registry to serve both EROFS and legacy
```

Comment:
I saw you had linked the issue to the OS features before. Not a big deal, but should we link it?